### ISSUE TITLE
Spack Needs to Check Python Version

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#1870]
Spack is intended to work only with Python2 at this time.  That is fine.  However, users occasionally run with Python3 by accident, which produces strange error messages:
```
$ python3 `which spack`
Traceback (most recent call last):
  File "/home2/rpfische/spack2/bin/spack", line 52, in <module>
    import nose
  File "/home2/rpfische/spack2/lib/spack/external/nose/__init__.py", line 1, in <module>
    from nose.core import collector, main, run, run_exit, runmodule
  File "/home2/rpfische/spack2/lib/spack/external/nose/core.py", line 153
    print "%s version %s" % (os.path.basename(sys.argv[0]), __version__)
                        ^
SyntaxError: invalid syntax
```

Spack needs to check Python version at the beginning of the `spack` script, and exit gracefully with an appropriate error message if it detects Python3.  Otherwise, it wastes users' time when they see this error.

This issue will become increasingly important as newer Linuxes (Ubuntu) ship with Python3 by default.

**Comment by adamjstewart** [comment]
Can we change the shebang to `#!/usr/bin/env python2`?

**Comment by citibeth** [comment]
Unfortunately no.  Won't work on a Mac, for example:
```
$ which python
/usr/bin/python
$ which python2
$
```

**Comment by tgamblin** [comment]
There is a version check at the top of `bin/spack`.  Unfortunately it only checks min version, and doesn't complain about 3.  Any takers for a PR?

**Comment by citibeth** [comment]
I'll do it

**Comment by adamjstewart** [comment]
Closed by #1872.
